packages/docs/docs/advanced/custom-components.mdx

```diff
@@ -0,0 +1,126 @@
+//  define
+
+export interface SwitchProps extends NodeProps {
+    initialState?: SignalValue<boolean>
+
+    // We dont use color here because we want
+    // to be able to pass hex strings and rgb
+    // values to accent rather than a `Color`
+    accent?: SignalValue<PossibleColor>
+}
+```
+Next, create a component class for your components. It must extend an existing
+component such as `Line`, `Rect` or `Layout`. If you dont want to inherit any
+methods from an existing component, extend your class from `Node`. It is advisable
+to use a component most similar to yours, for instance if you were to make a component
+including a `Layout`, you should extend `Layout` and `LayoutProps`.
+```tsx
+export interface SwitchProps extends NodeProps { ... }
+
+export class Switch extends Node { ... }
+```
+
+To use the properties defined in the interface, your class _must_ contain a property with the
+same name. Motion Canvas provides type decorators to facilitate this. Here is an example of
+how you would define such a property:
+```tsx
+export class Switch extends Node {
+    // @initial - sets the property to an
+    // initial value if it was not provided.
+    @initial(false)
+    // @signal - is required by motion canvas
+    // for every prop that was passed in.
+    @signal()
+    public declare readonly initialState: SimpleSignal<boolean, this>
+
+    @initial("#68ABDF")
+    @signal()
+    public declare readonly accent: ColorSignal<this>
+```
+Notice how colors are wrapped in `ColorSignal<>` whilst any other type (even user-defined ones)
+are wrapped in `SimpleSignal<>`. The type does not need to be passed to color signal as Motion
+Canvas knows that it must be of a color-resolvable type. In both, the class is passed at the end
+of the wrapper to register the signal to the class. The properties must also be _declared_, _public_ 
+and _readonly_.
+
+Normal properties of a component class can be defined as normal. For example like so:
+```tsx
+private isOn: boolean;
+```
+
+Every component _requires_ a constructor. The format of the constructor will always
+look similar to this, with optional argument for the properties:
+```tsx
+public constructor(props?: SwitchProps) {
+    super({
+        ...props
+    })
+    ...
+}
+```
+Now the properties will follow the schema defined in your interface and they will be
+passed to the superclass, in our case `Node`. The `props` parameter can also be useful
+outside of the super call for instance, setting styles on every rect in an array.
+
+Now we can add elements to the view by using `this.add()`, much like you would add to a
+scene's view:
+```tsx
+public constructor(props?: SwitchProps) {
+    ...
+    this.add(
+        <>
+            <Rect
+                ...
+            >
+                <Circle ... />
+            </Rect>
+        </>
+    )
+}
+```
+Since this is still a regular typescript class, you can add methods. This is especially useful
```

Comment:
nit: "Since this is still a regular typescript class, you can also add methods."

"Since this is still..." normally implies some base level of information, so "also" makes sense here.